public/app/features/variables/state/sharedReducer.ts

```diff
@@ -74,9 +74,18 @@ const sharedReducerSlice = createSlice({
     },
     duplicateVariable: (state: VariablesState, action: PayloadAction<VariablePayload<{ newId: string }>>) => {
       const original = cloneDeep<VariableModel>(state[action.payload.id]);
-      const name = `copy_of_${original.name}`;
-      const newId = action.payload.data?.newId ?? name;
+      let name = `copy_of_${original.name}`;
+      const originalId = action.payload.data?.newId ?? name;
+      let newId = action.payload.data?.newId ?? name;
```

Comment:
This is confusing, original and new ID are the same

Reply:
This is confusing, original and new ID are the same; To make it clear, I'd suggest doing:
```suggestion
      const originalId = action.payload.data?.newId ?? name;
      let newId = originalId;
```